lib/rubocop/cop/processed_source_logic.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # This module extends ProcessedSource with helper functionality.
+    module ProcessedSourceLogic
```

Comment:
I can't decide whether it nice to separate this logic into a module, or to just include it in `ProcessedSource` directly. Also, I am open to suggestions if someone feels strongly about a different name.

Reply:
A mixin that's extremely class-specifix is not really a mixin IMO. I'd put everything in `ProcessedSource` probably (at least pretty common methods). I see here and there methods that seem like something that's used just once or twice and I wonder what to do with them. Perhaps its fine for such cases to just keep this logic in the relevant cops.

Reply:
> A mixin that's extremely class-specific is not really a mixin IMO.

Agreed. My reasoning for going this route was the way we include `AutocorrectLogic` into our `Cop` class.  In a separate PR, I can move these methods directly into `cop.rb`.

> I'd put everything in ProcessedSource probably (at least pretty common methods). I see here and there methods that seem like something that's used just once or twice and I wonder what to do with them. Perhaps its fine for such cases to just keep this logic in the relevant cops.

I'm torn about this. 

I think the one I listed in the issue, `aligned_comments?`, is the worst example here. It seems scoped to alignment issues and thus does seem best placed back into its alignment mixin. `empty_brackets?` looks like it should be moved back as well. These two methods required very specific args. One has to be a comment token, and the latter has to be opening and closing bracket tokens. Reuse seems unlikely.

But the rest of them (the ones currently in the Module), although only used once or twice, seem more like the responsible of a `processed_source` and not a cop/mixin. The args for these are more generic (a token, a range, an index, a line, a string).

I'll make these alterations. Feel free to nudge me again to move more back :) .

Reply:
Hmm.  `previous_and_current_lines_empty?` and `stripped_upto?` also seem to have a narrow use case. I'll move these back too.